### PR TITLE
Integrate mode switch for automated tests

### PR DESCRIPTION
**Pull request by CadeFaulkner**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PO-118
https://tools.hmcts.net/jira/browse/PO-120
https://tools.hmcts.net/jira/browse/PO-129
https://tools.hmcts.net/jira/browse/PO-131

### Change description ###

adding the ability to test both legacy and opal routes
adding tests for the above tickets

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by CadeFaulkner** [comment]
Current failure valid, waiting on fix for PO-120

**Comment by CadeFaulkner** [comment]
@sabahirfan the legacy-db stub doesn't look to be running in the dev pipelines, so all the legacy tests are failing

File: src/functionalTest/java/uk/gov/hmcts/opal/steps/ModeSwitchStepDef.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+public class ModeSwitchStepDef extends BaseStepDef {
+    @BeforeAll
+    public static void beforeAll() throws JSONException {
+        String testMode = System.getProperty("test.mode");
+
+        if (Objects.equals(testMode, "opal")) {
+            JSONObject requestBody = new JSONObject();
+            requestBody.put("mode", "opal");
+
+            System.out.println("Test mode equals " + testMode + " setting to: opal");
```
Comment (sabahirfan) [inline src/functionalTest/java/uk/gov/hmcts/opal/steps/ModeSwitchStepDef.java:20]: let's use `log.info` instead of `System.out`

**Review by sabahirfan — APPROVED** [review]
LGTM

**Pull request merged**